hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/bucket/BucketCache.java

```diff
@@ -1053,6 +1066,7 @@ private void persistToFile() throws IOException {
     }
     try (FileOutputStream fos = new FileOutputStream(persistencePath, false)) {
       fos.write(ProtobufMagic.PB_MAGIC);
+      checksum = ((PersistentIOEngine) ioEngine).calculateChecksum(algorithm);
       BucketProtoUtils.toPB(this).writeDelimitedTo(fos);
```

Comment:
The checksum can be passed as an argument.  Keeping checksum as a state var in BC does not look correct.